06-building-a-program-start-to-finish/writing-tests/output/spec/invertedSpec.js

```diff
@@ -0,0 +1,257 @@
+        },
+
+        {
+          title: 'The Lord of the Rings: The Fellowship of the Ring.',
+          text: 'an unusual alliance of man, elf, dwarf, wizard and hobbit seek to destroy a powerful ring.',
+        },
+      ];
+      expect(newIndex.lowerDocText(lowerDocFile)).toEqual(result);
+    });
+  });
+
+  describe('Tests for the Search methods', () => {
+    const docJsonMock = {'doc.json': [{ alice: [0],
+      alliance: [1],
+      an: [1],
+      man: [1],
+      hole: [0],
+      a: [0],
+      falls: [0],
+      rabbit: [0],
+      unusual: [1] }, ['Alice in Wonderland', 'The Lord of the Rings: The Fellowship of the Ring.']]};
+
+    it('should return an object of words that appear in a single indexed file', () => {
+      const result = { alice: [0],
+        alliance: [1],
+        an: [1],
+        man: [1],
+        hole: [0],
+        a: [0],
+        falls: [0],
+        rabbit: [0],
+        unusual: [1] };
+      expect(newIndex.searchIndex('doc.json', 'alice,an,hole a man', docJsonMock))
+          .toEqual(result);
+    });
+
+    it('the buildSearchResult should build and return correct search result', () => {
+      const result = { alice: [0],
+        alliance: [1],
+        an: [1],
+        man: [1],
+        hole: [0],
+        a: [0],
+        falls: [0],
+        rabbit: [0],
+        unusual: [1] };
+        expect(newIndex.buildSearchResult('doc.json', 'alice,an,hole a man', docJsonMock))
+            .toEqual(result);
+    });
+
+    it('returns an object of file names against their search results for a search of all files', () => {
+      const allFilesResult = newIndex.searchIndex('All Files', 'alice,an,hole,a - man', allFilesMock);
+      const result = { 'doc.json': {
+        alice: [0],
+        alliance: [1],
+        an: [1],
+        man: [1],
+        hole: [0],
+        a: [0],
+        falls: [0],
+        rabbit: [0],
+        unusual: [1] },
+
+        'test.json': { alice: [0],
+          alliance: [1, 2],
+          an: [1, 2],
+          and: [0, 1, 2],
+          man: [1, 2],
+          hole: [0],
+          a: [0, 1, 2],
+          dwarf: [1, 2],
+          falls: [0],
+          imagination: [0],
+          rabbit: [0],
+          unusual: [1, 2],
+          wizard: [1, 2] } };
+      expect(allFilesResult).toEqual(result);
+    });
+
+    it('should take search string in any case and return result', () => {
+      const result = {
+        alice: [0],
+        alliance: [1],
+        an: [1],
+        man: [1],
+        hole: [0],
+        a: [0],
+        falls: [0],
+        rabbit: [0],
+        unusual: [1] };
+      expect(newIndex.searchIndex('doc.json', 'ALICE,AN,HOLE A MAN', docJsonMock)).toEqual(result);
+    });
+
+    it('should return words containing substring in search query', () => {
+      const result = { alliance: [1],
+        an: [1],
+        man: [1] };
+      expect(newIndex.searchIndex('doc.json', 'an', docJsonMock)).toEqual(result);
+    });
+  });
+});
```

Comment:
I have fixed all these issues now, except the issue of undefined variables. The variables being flagged as undefined are variables I am using from another file.